### PR TITLE
pt-BR translation for RPC-Pay

### PR DESCRIPTION
**Pull request by botoserelepe**
Hey @lh1008. Here's RPC-Pay Up Close translation.

Ping @netrik182 and @nelsonrenan for review.

### CODE REVIEW

**Comment by netrik182** [comment]
Reviewing it right now.

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# RPC - Pay de perto
+
+**_"Ignorado e pouco valorizado até então, o RCP-Pay permite microtransações instantâneas e privadas e é a mais nova menor unidade de pagamento no universo Monero."_ - 29 de Janeiro de 2020**
+
+Por que rodar um nó público de Monero? Gera gastos, mas não gera dinheiro - pelo menos, não diretamente. Um a preocupação recorrente é como incentivar mais desses nós. O RPC-Pay oferece esse incentivo e muito mais. Ele fornece uma maneira de usar hashes de mineração de Monero - apenas os hashes - para pagar as Chamadas de Procedimento Remoto (do inglês Remote Procedure Calls, RPC) de qualquer servidor habilitado para o RPC-Pay, como, por exemplo, um nó de Monero. As hashes recebidas como pagamento são usadas pelo servidor para ganhar dinheiro e, devido ao RandomX do Monero, as hashes de mineração podem ser calculadas de maneira eficiente usando as CPUs mais populares. Pagar apenas com hashes em vez de Monero em si significa que o RPC-Pay não sobrecarrega a rede Monero e nem deixa registros. Com o RPC-Pay, as hashes de mineração se tornam a mais nova menor unidade de pagamento no universo Monero.
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:5]:
```suggestion
Por que rodar um nó público do Monero? Gera gastos, mas não gera dinheiro - pelo menos, não diretamente. Uma preocupação recorrente é como incentivar mais desses nós. O RPC-Pay oferece esse incentivo e muito mais. Ele fornece uma maneira de usar hashes de mineração de Monero - apenas os hashes - para pagar as Chamadas de Procedimento Remoto (do inglês Remote Procedure Calls, RPC) de qualquer servidor habilitado para o RPC-Pay, como, por exemplo, um nó do Monero. As hashes recebidas como pagamento são usadas pelo servidor para ganhar dinheiro e, devido ao RandomX do Monero, as hashes de mineração podem ser calculadas de maneira eficiente usando as CPUs mais populares. Pagar apenas com hashes em vez de Monero em si significa que o RPC-Pay não sobrecarrega a rede Monero e nem deixa registros. Com o RPC-Pay, as hashes de mineração se tornam a mais nova menor unidade de pagamento no universo Monero.
```

My suggestion is to change "de" to "do" in some places.

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+### _RPC_
+
+Antes de estudar o RPC-Pay, é importante entender o vasto conceito de RPC. RPC é o processo pelo qual um programa (um cliente) solicita dados e cálculos de outro programa (um servidor). O servidor pode estar em qualquer computador conectado à Internet. Uma solicitação RPC é como uma chamada de sub-rotina em um programa comum, exceto que, com o RPC, o trabalho da sub-rotina é feito no servidor. Softwares intermediários especiais lidam com a comunicação de rede de ambas solicitação e resposta, de forma que seja imperceptível ao programador. Muitos tipos diferentes de interface de software têm sido usadas para implementar o RPC ao longo dos anos.
+
+Uma família moderna de interface RPC usa a linguagem JavaScript Object Notation (JSON) para formar mensagens. A JSON é um formato textual padrão para câmbio de dados. Ela depende de pares nome-valor e listas de dados para definir a informação de uma maneira que seja robusta, extensível e capaz de ser lida por humanos. A codificação JSON  é o método primário usado pela estrutura do RPC do Monero.
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:11]:
```suggestion
Uma família moderna de interface RPC usa a linguagem JavaScript Object Notation (JSON) para formar mensagens. A JSON é um formato textual padrão para câmbio de dados. Ela depende de pares nome-valor e listas de dados para definir a informação de uma maneira que seja robusta, extensível e capaz de ser lida por humanos. A codificação JSON é o método primário usado pela estrutura do RPC do Monero.
```

Removed doubled blank "spaces"

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+Para Monero, o RPC já está sendo usado em uma variedade de maneiras. Por exemplo, o daemon padrão do Monero, contendo a maioria das funcionalidades do Monero, é o _monerod_. Ele implementa a parte de servidor do RPC para uma variedade de chamadas de funções relacionadas a blockchain. E o programa _monero-wallet-rpc_, entregue com a instalação padrão do Monero, implementa a parte de servidor de funcionalidades de carteira enquanto ele próprio é cliente de um servidor _monerod_. O RPC do Monero chama trabalho muito como um navegador comunicando com um servidor da web. Os dados são enviados usando o método POST, e a informação devolvida é textual, geralmente em JSON. Se você estiver interessado em exemplos detalhados de mensagens RPC, por favor, veja o [Exemplo #1](https://www.monerooutreach.org/stories/RPC-Pay.html#box1) em como usar chamadas RPC do Monero.
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:13]:
```suggestion
No Monero, o RPC já está sendo usado em uma variedade de maneiras. Por exemplo, o daemon padrão do Monero, contendo a maioria das funcionalidades do Monero, é o _monerod_. Ele implementa a parte de servidor do RPC para uma variedade de chamadas de funções relacionadas à blockchain. E o programa _monero-wallet-rpc_, entregue com a instalação padrão do Monero, implementa a parte de servidor de funcionalidades de carteira enquanto ele próprio é cliente de um servidor _monerod_. As chamadas RPC do Monero funcionam como um navegador comunicando com um servidor da web. Os dados são enviados usando o método POST, e a informação devolvida é textual, geralmente em JSON. Se você estiver interessado em exemplos detalhados de mensagens RPC, por favor, veja o [Exemplo #1](https://www.monerooutreach.org/stories/RPC-Pay.html#box1) em como usar chamadas RPC do Monero.
```

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+## _Hashes de mineração_
+
+O processo de mineração para a Prova de Trabalho do RandomX do Monero começa executando um programa pseudo-aleatório que é a função de novas transações, um nonce, um endereço de pagamento, e dados de um bloco anterior. Executar um programa aleatório como esse é fácil para a CPU, mas difícil para ASICs. A saída do programa passa por uma função criptografada de hash, que o transforma em um número de 256-bit. Se esse número é menor que o limite especificado, então conta-se como uma mineração de um novo bloco.
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:17]:
```suggestion
O processo de mineração para a Prova de Trabalho do RandomX do Monero começa executando um programa pseudo-aleatório que é uma função das novas transações, de um nonce, de um endereço de pagamento, e de dados de um bloco anterior. Executar um programa aleatório como esse é fácil para a CPU, mas difícil para ASICs. A saída do programa passa por uma função criptografada de hash, que o transforma em um número de 256-bit. Se esse número é menor que o limite especificado, então conta-se como uma mineração de um novo bloco.
```

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+Uma beleza desse processo de hash é que o endereço de pagamento é especificado antes da hash ser gerada, o que permite que hashes sejam criadas especificamente por outra pessoa. Essa propriedade permite que as pools de mineração funcionem - hashes são feitas usando os endereços de pagamento da pool. Existem poucas analogias da hash de mineração com o mundo físico. É uma criação que só pode ter valor para o dono do endereço de destino, e esse valor dura até que o próximo bloco seja minerado. Os blocos de Monero são minerados a cada 20 minutos aproximadamente.
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:19]:
```suggestion
Uma beleza desse processo de hash é que o endereço de pagamento é especificado antes da hash ser gerada, o que permite que hashes sejam criadas especificamente para outra pessoa. Essa propriedade permite que as pools de mineração funcionem - hashes são feitas usando os endereços de pagamento da pool. Existem poucas analogias da hash de mineração com o mundo físico. É uma criação que só pode ter valor para o dono do endereço de destino, e esse valor dura até que o próximo bloco seja minerado. Os blocos do Monero são minerados a cada 2 minutos aproximadamente.
```

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+## _RPC-Pay_
+
+RPC-Pay é um novo recurso adicionado ao _monerod_ e às carteiras que habilita o pagamento para chamadas RPC usando hashes de mineração. Essa capacidade é configurada através da linha de comando quando o software do _monerod_ está sendo inciado. Parâmetros de inicialização representativos estão ilustrados abaixo no [Exemplo #2](https://www.monerooutreach.org/stories/RPC-Pay.html#box2). De forma similar, existem comandos com a carteira CLI que habilitam a criação de hashes e pagam _monerod_. Esses comandos também estão apresentados no [Exemplo #2](https://www.monerooutreach.org/stories/RPC-Pay.html#box2).
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:25]:
```suggestion
RPC-Pay é um novo recurso adicionado ao _monerod_ e às carteiras que habilita o pagamento para chamadas RPC usando hashes de mineração. Essa capacidade é configurada através da linha de comando quando o software do _monerod_ está sendo inciado. Parâmetros de inicialização representativos estão ilustrados abaixo no [Exemplo #2](https://www.monerooutreach.org/stories/RPC-Pay.html#box2). De forma similar, existem comandos com a carteira CLI que habilitam a criação de hashes e o pagamento ao _monerod_. Esses comandos também estão apresentados no [Exemplo #2](https://www.monerooutreach.org/stories/RPC-Pay.html#box2).
```

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+O RPC-Pay é privado. Nenhum grande balanço de crédito que exija compartilhamento de informação para backup ou segurança é mantido. As hashes de mineração usadas como pagamento são anônimas, sem uma forma de rastrear o proprietário na blockchain. Uma identidade de pagamento pode ser aplicada para acompanhar e manter balanços, mas com um procedimento de operação baseado em balanços pequenos, essa identidade pode ser mudada regularmente ou nem sequer usada.
+
+É importante considerar também as desvantagens da atual implementação _monerod_/carteira do RPC-Pay. Isso pode sobrecarregar clientes com baixo poder, como dispositivos móveis, os quais dependem de outros computadores ajudantes para suas hashes. Ademais, o rendimento para o servidor é aleatório. Ele vem somente quando o cliente fornece uma hash capaz de minerar um bloco, enquanto a saída do servidor deve ser contínua, casando com a provisão regular de hashes. Finalmente, a quantidade de renda que um servidor pode ter é limitada pela agenda de mineração do Monero e tenderá a ser pequena.
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:33]:
```suggestion
É importante considerar também as desvantagens da atual implementação _monerod_/carteira do RPC-Pay. Isso pode sobrecarregar clientes com baixo poder de processamento, como dispositivos móveis, os quais dependem de outros computadores ajudantes para suas hashes. Ademais, o rendimento para o servidor é aleatório. Ele vem somente quando o cliente fornece uma hash capaz de minerar um bloco, enquanto a saída do servidor deve ser contínua, casando com a provisão regular de hashes. Finalmente, a quantidade de renda que um servidor pode ter é limitada pela agenda de mineração do Monero e tenderá a ser pequena.
```

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+Tem mais sobre o RPC-Pay que apenas programas Monero pagando outros programas Monero, já que qualquer servidor consegue usar o RPC-Pay para gerar renda. O software do nó oficial do Monero e da carteira é de código aberto e pode ser reutilizado para aplicações inovadoras, ou outras aplicações podem se comunicar com as instâncias _monerod_. O conceito fundamental é simplesmente que o servidor dá seu endereço Monero para seus clientes e seus clientes usam isso para criar hashes de mineração Monero que são mandadas ao servidor como pagamento. O Exemplo #3 dá um exemplo de código-fonte de software de RPC-Pay.
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:35]:
```suggestion
Tem mais sobre o RPC-Pay que apenas programas Monero pagando outros programas Monero, já que qualquer servidor consegue usar o RPC-Pay para gerar renda. O software do nó oficial do Monero e da carteira é de código aberto e pode ser reutilizado para aplicações inovadoras, ou outras aplicações podem se comunicar com as instâncias _monerod_. O conceito fundamental é simplesmente que o servidor dá seu endereço Monero para seus clientes e seus clientes usam isso para criar hashes de mineração Monero que são mandadas ao servidor como pagamento. O [Exemplo #3](https://www.monerooutreach.org/stories/RPC-Pay.html#box3) dá um exemplo de código-fonte de software de RPC-Pay.
```

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+## _Primo_
+
+Um exemplo de aplicação do novo RPC-Pay é o projeto Primo (link). Primo é um protocolo e pacote de software que suporta a entrega de conteúdo de sites pagos. Ele permite que anúncios indesejados nos sites sejam substituídos pela geração invisível e driscreta de hashes. Caso um visitante de um site habilitado para o Primo não queira ver anúncios, ele pode optar por, no lugar dos anúncios, calcular hashes de mineração Monero.
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:39]:
```suggestion
Um exemplo de aplicação do novo RPC-Pay é o projeto Primo [(repo.getmonero.org/selene/primo)](https://repo.getmonero.org/selene/primo). Primo é um protocolo e pacote de software que suporta a entrega de conteúdo de sites pagos. Ele permite que anúncios indesejados nos sites sejam substituídos pela geração invisível e driscreta de hashes. Caso um visitante de um site habilitado para o Primo não queira ver anúncios, ele pode optar por, no lugar dos anúncios, calcular hashes de mineração Monero.
```

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+O Primo tem 3 componentes. O primeiro é o módulo _primo-apache_ para uso no servidor web. O dono do site instala esse módulo e configura quais conteúdos exigem pagamento e quanto deve ser pago. O servidor web se comunica com a instância _monerod_, que lida com a contabilidade os pagamentos de hashes e usa hashes bem-sucedidas para coletar a arrecadação da mineração. O segundo componente é a extensão _primo-firefox_. Um visitante do site que queira usar RPC-Pay terá ela instalada no seu navegador Firefox. E o terceiro componente é a ferramenta de controle gráfico _primo-control-center_. Com ela, o usuário do Firefox pode configurar quais sites recebem pagamento.
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:41]:
```suggestion
O Primo tem 3 componentes. O primeiro é o módulo _primo-apache_ para uso no servidor web. O dono do site instala esse módulo e configura quais conteúdos exigem pagamento e quanto deve ser pago. O servidor web se comunica com a instância _monerod_, que lida com a contabilidade dos pagamentos de hashes e usa hashes bem-sucedidas para coletar a arrecadação da mineração. O segundo componente é a extensão _primo-firefox_. Um visitante do site que queira usar RPC-Pay terá ela instalada no seu navegador Firefox. E o terceiro componente é a ferramenta de controle gráfico _primo-control-center_. Com ela, o usuário do Firefox pode configurar quais sites recebem pagamento.
```

File: monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+O número de blocos na blockchain quando isso foi rodado era de 2.021.560.
+
+Se você estiver rodando sua própria instância _monerod_ com uma configuração padrão, você conseguirá a mesma informação com o seguinte comando, muito similar, no mesmo computador:
+
+```
+curl -X POST [127.0.0.1:18081/json_rpc](http://127.0.0.1:18081/json_rpc) -H 'Content-Type:application/json' --data "{\"jsonrpc\":\"2.0\",\"id\":\"0\",\"method\":\"get_block_count\"}"
+```
+
+Para mais informações sobre a interface _monerod_ RPC, veja o [www.getmonero.org/resources/developer-guides/daemon-rpc.html](https://www.getmonero.org/resources/developer-guides/daemon-rpc.html). E para informações sobre a interface carteira RPC, veja [www.getmonero.org/resources/developer-guides/wallet-rpc.html](https://www.getmonero.org/resources/developer-guides/wallet-rpc.html).
```
Comment (netrik182) [inline monero-outreach-docs/translations/pt-br/archive/rpc_pay_pt-br.md:74]:
```suggestion
Para mais informações sobre a interface _monerod_ RPC, veja o [www.getmonero.org/resources/developer-guides/daemon-rpc.html](https://www.getmonero.org/pt-br/resources/developer-guides/daemon-rpc.html). E para informações sobre a interface carteira RPC, veja [www.getmonero.org/resources/developer-guides/wallet-rpc.html](https://www.getmonero.org/pt-br/resources/developer-guides/wallet-rpc.html).
```

Both guides are not translated yet in getmonero.org but might be a good idea to link to them anyway.

**Review by netrik182 — CHANGES_REQUESTED** [review]
Very solid translation once againg @botoserelepe.

I've added some suggestions to your translation and links where you missed it.

**Comment by botoserelepe** [comment]
@lh1008 I changed what was requested by @netrik182 . You can merge it if you agree with it. :)

**Review by netrik182 — APPROVED** [review]
Looks good @botoserelepe. Thank you!

**Review by lh1008 — APPROVED** [review]
Everything looks good. Thank you @netrik182 and @botoserelepe.

:)

**Pull request merged**